### PR TITLE
[SPARK-45923]Add java21 to the e2e tests

### PR DESCRIPTION
**Pull request by TQJADE**
### What changes were proposed in this pull request?
Add java 21 e2e teste

### Why are the changes needed?
E2E coverage on java version

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
e2e workflow should cover this

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

File: tests/e2e/spark-versions/chainsaw-test.yaml [hunk 1]
```diff
@@ -47,7 +47,16 @@ spec:
...
+        value: 'apache/spark:3.4.3-scala2.12-java11-ubuntu'
+  - bindings:
+      - name: "SPARK_VERSION"
+        value: "4.0.0-preview1"
+      - name: "SCALA_VERSION"
+        value: "2.13"
+      - name: "JAVA_VERSION"
+        value: "21"
+      - name: "IMAGE"
+        value: 'apache/spark:4.0.0-preview1-java21-scala'
```
Comment (TQJADE) [inline tests/e2e/spark-versions/chainsaw-test.yaml:59]: Same image:
https://github.com/apache/spark-kubernetes-operator/blob/b18776458b8f26db8e43f6ddb9275f3cbdbafffe/examples/pi-java21.yaml#L28

File: tests/e2e/spark-versions/chainsaw-test.yaml [hunk 2]
```diff
@@ -38,7 +38,7 @@ spec:
       - name: "JAVA_VERSION"
         value: "17"
       - name: "IMAGE"
-        value: 'spark:3.5.2-scala2.12-java17-ubuntu'
+        value: 'apache/spark:3.5.2-scala2.12-java17-ubuntu'
```
Comment (dongjoon-hyun) [inline tests/e2e/spark-versions/chainsaw-test.yaml:41]: Thank you!

File: tests/e2e/spark-versions/chainsaw-test.yaml [hunk 3]
```diff
@@ -47,7 +47,16 @@ spec:
       - name: "JAVA_VERSION"
         value: "11"
       - name: "IMAGE"
-        value: 'spark:3.4.3-scala2.12-java11-ubuntu'
+        value: 'apache/spark:3.4.3-scala2.12-java11-ubuntu'
```
Comment (dongjoon-hyun) [inline tests/e2e/spark-versions/chainsaw-test.yaml:50]: Thank you.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
To @TQJADE , please be careful when you write the PR title with JIRA ID.

**Pull request closed without merging**